src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,371 @@
+
+        return new google.payments.api.PaymentsClient(options);
+    }
+
+    private _setExternalCheckoutData(paymentData: GooglePaymentData): Promise<void> {
+        return this._googlePayInitializer.parseResponse(paymentData)
+            .then((tokenizePayload: TokenizePayload) => {
+                const paymentSuccessPayload: PaymentSuccessPayload = {
+                    tokenizePayload,
+                    billingAddress: paymentData.cardInfo.billingAddress,
+                    shippingAddress: paymentData.shippingAddress,
+                    email: paymentData.email,
+                };
+
+                const {
+                    onError = () => {},
+                    onPaymentSelect = () => {},
+                } = this._googlePayOptions;
+
+                this._updateShippingAndBillingAddress(paymentSuccessPayload).then(() => {
+                    return this._paymentInstrumentSelected(paymentSuccessPayload)
+                        .then(() => onPaymentSelect())
+                        .catch(error => onError(error));
+                });
+            });
+    }
+
+    private _paymentInstrumentSelected(paymentSuccessPayload: PaymentSuccessPayload): Promise<InternalCheckoutSelectors> {
+        if (!this._paymentMethod) {
+            throw new Error('Payment method not initialized');
+        }
+
+        const { id: methodId } = this._paymentMethod;
+
+        return this._store.dispatch(this._paymentStrategyActionCreator.widgetInteraction(() => {
+            return this._postForm(paymentSuccessPayload)
+                .then(() => Promise.all([
+                    this._store.dispatch(this._checkoutActionCreator.loadCurrentCheckout()),
+                    this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId)),
+                ]));
+        }, { methodId }), { queueId: 'widgetInteraction' });
+    }
+
+    private _postForm(paymentData: PaymentSuccessPayload): Promise<Response<any>> {
+        const cardInformation = paymentData.tokenizePayload.details;
+
+        return this._requestSender.post('/checkout.php', {
+            headers: {
+                Accept: 'text/html',
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            body: toFormUrlEncoded({
+                payment_type: paymentData.tokenizePayload.type,
+                nonce: paymentData.tokenizePayload.nonce,
+                provider: this._methodId,
+                action: 'set_external_checkout',
+                card_information: this._getCardInformation(cardInformation),
+            }),
+        });
+    }
+
+    private _getCardInformation(cardInformation: { cardType: string, lastFour: string }) {
+        return {
+            type: cardInformation.cardType,
+            number: cardInformation.lastFour,
+        };
+    }
+
+    private _handleError(error: Error): never {
+        throw new StandardError(error.message);
```

Comment:
We could probably just throw the error directly instead of having this extra method.